azure/sdk-identity-resources-storage/Program.cs

```diff
@@ -0,0 +1,181 @@
+
+            // Azure.ResourceManager.Resources is currently in preview.
+            var resourcesManagementClient = new ResourcesManagementClient(subscriptionId, credential);
+
+            // Azure.ResourceManager.Storage is currently in preview.
+            var storageManagementClient = new StorageManagementClient(subscriptionId, credential);
+
+            try
+            {
+                // Create a Resource Group
+                string resourceGroupName = await CreateResourceGroupAsync(resourcesManagementClient);
+
+                // Create a Storage account
+                string storageName = await CreateStorageAccountAsync(storageManagementClient, resourceGroupName);
+
+                // Create a container and upload a blob using a storage connection string
+                await UploadBlobUsingStorageConnectionStringAsync(storageManagementClient, resourceGroupName, storageName);
+
+                // Upload a blob using Azure.Identity.DefaultAzureCredential
+                await UploadBlobUsingDefaultAzureCredentialAsync(storageManagementClient, resourceGroupName, storageName, credential);
+
+
+                Console.WriteLine("Press any key to continue and delete the resources...");
+                Console.ReadKey(true);
+
+                // Delete the resource group
+                await DeleteResourceGroupAsync(resourcesManagementClient, resourceGroupName);
+            }
+            catch (RequestFailedException ex)
+            {
+                Console.WriteLine($"Request failed! {ex.Message} {ex.StackTrace}");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Unexpected exception! {ex.Message} {ex.StackTrace}");
+            }
+        }
+
+        private static async Task<string> CreateResourceGroupAsync(ResourcesManagementClient resourcesManagementClient)
+        {
+            string resourceGroupName = RandomName("rg", 20);
+            Console.WriteLine($"Creating resource group {resourceGroupName}...");
+            await resourcesManagementClient.ResourceGroups.CreateOrUpdateAsync(resourceGroupName, new ResourceGroup(ResourceRegion));
+            Console.WriteLine("Done!");
+
+            return resourceGroupName;
+        }
+
+        private static async Task<string> CreateStorageAccountAsync(StorageManagementClient storageManagementClient, string rgName)
+        {
+            Console.WriteLine("Creating a new storage account...");
+
+            string storageAccountName = await GenerateStorageAccountNameAsync(storageManagementClient);
+            StorageAccountCreateParameters parameters = new StorageAccountCreateParameters(
+                                                    new Sku("Standard_LRS"),
+                                                    Kind.BlobStorage,
+                                                    ResourceRegion)
+                                                    { AccessTier = AccessTier.Hot };
```

Comment:
It looks like this line won't compile.

Reply:
Why do you say that? The indents are kinda weird because I couldn't decide how to make it the most readable but it compiles and runs.

Reply:
Disregard. It displays properly for me in diff view now.